Test failure:
```
14:49:51 [ERROR] Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.849 s <<< FAILURE! - in com.yahoo.vespa.hosted.controller.maintenance.JobRunnerTest
14:49:51 [ERROR] multiThreadedExecutionFinishes(com.yahoo.vespa.hosted.controller.maintenance.JobRunnerTest)  Time elapsed: 0.015 s  <<< ERROR!
14:49:51 java.lang.IllegalArgumentException: Cannot start systemTest for tenant.real; it is already running!
14:49:51 	at com.yahoo.vespa.hosted.controller.deployment.JobController.lambda$start$44(JobController.java:484)
14:49:51 	at com.yahoo.vespa.hosted.controller.deployment.JobController.locked(JobController.java:607)
14:49:51 	at com.yahoo.vespa.hosted.controller.deployment.JobController.start(JobController.java:481)
14:49:51 	at com.yahoo.vespa.hosted.controller.deployment.JobController.start(JobController.java:476)
14:49:51 	at com.yahoo.vespa.hosted.controller.deployment.JobController.start(JobController.java:471)
14:49:51 	at com.yahoo.vespa.hosted.controller.maintenance.JobRunnerTest.multiThreadedExecutionFinishes(JobRunnerTest.java:98)
```

--- a/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java
+++ b/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java
@@ -3,7 +3,6 @@ package com.yahoo.vespa.hosted.controller.deployment;
 
 import com.google.common.collect.ImmutableSortedMap;
 import com.yahoo.component.Version;
-import com.yahoo.config.application.api.DeploymentSpec;
 import com.yahoo.config.provision.ApplicationId;
 import com.yahoo.config.provision.zone.ZoneId;
 import com.yahoo.vespa.curator.Lock;
@@ -482,7 +481,7 @@ public class JobController {
         locked(id, type, __ -> {
             Optional<Run> last = last(id, type);
             if (last.flatMap(run -> active(run.id())).isPresent())
-                throw new IllegalStateException("Can not start " + type + " for " + id + "; it is already running!");
+                throw new IllegalArgumentException("Cannot start " + type + " for " + id + "; it is already running!");
 
             RunId newId = new RunId(id, type, last.map(run -> run.id().number()).orElse(0L) + 1);
             curator.writeLastRun(Run.initial(newId, versions, isRedeployment, controller.clock().instant(), profile));

--- a/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunnerTest.java
+++ b/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunnerTest.java
@@ -98,7 +98,7 @@ public class JobRunnerTest {
             jobs.start(id, systemTest, versions);
             fail("Job is already running, so this should not be allowed!");
         }
-        catch (IllegalStateException ignored) { }
+        catch (IllegalArgumentException ignored) { }
         jobs.start(id, stagingTest, versions);
 
         assertTrue(jobs.last(id, systemTest).get().stepStatuses().values().stream().allMatch(unfinished::equals));

